### ISSUE TITLE
php warnings after plugin-activation on plugins screen in German install

### ISSUE THREAD

**Issue by wdfee** [woothemes/woocommerce#38191]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Right after re-activating woocommerce in a German install appear the following php warnings, when debug mode is set to true (that is in my case, because it's my local development environment):
`
Warning: Undefined array key “comment” in /[…]/plugins/woocommerce/src/Internal/Admin/Translations.php on line 84
Warning: Trying to access array offset on value of type null in /[…]/plugins/woocommerce/src/Internal/Admin/Translations.php on line 84
`

I reported this in the support forum: https://wordpress.org/support/topic/php-warning-in-7-6-0-dashboard-debug-mode/
Yes, I already tested it in a fresh install, too. 
I'll post my debugging results as first answer below.

### Expected behavior

I expect no warnings to appear.

### Actual behavior

Right after activating woocommerce on top of the loaded plugins screen appear two described php warnings.

### Steps to reproduce

1. install WordPress with German language pack (set site language to German) and debug mode on
2. install WooCommerce and activate it, set Storefront theme in setup wizard
3. go to plugins screen and deactivate woocommerce
4. activate woocommerce
5. the warnings appear on top of the loaded page

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://[...] [cutted this out, because it's local]
Site address (URL): https://[...]
WC Version: 7.6.1
REST API Version: ✔ 7.6.1
WC Blocks Version: ✔ 9.8.5
Action Scheduler Version: ✔ 3.5.4
Log Directory Writable: ✔
WP Version: 6.2
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: de_DE
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.54 (Unix) OpenSSL/1.1.1s PHP/8.0.27
PHP Version: 8.0.27
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.87.0
(SecureTransport) OpenSSL/1.1.1s

SUHOSIN Installed: –
MySQL Version: 8.0.31
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 7.6.1
WC Database Prefix: gg_
Datenbank-Gesamtgröße: 4.88MB
Datenbank-Datengröße: 3.42MB
Datenbank-Indexgröße: 1.46MB
gg_woocommerce_sessions: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
gg_woocommerce_api_keys: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
gg_woocommerce_attribute_taxonomies: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
gg_woocommerce_downloadable_product_permissions: Daten: 0.02MB + Index: 0.06MB + Engine InnoDB
gg_woocommerce_order_items: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
gg_woocommerce_order_itemmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
gg_woocommerce_tax_rates: Daten: 0.02MB + Index: 0.06MB + Engine InnoDB
gg_woocommerce_tax_rate_locations: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
gg_woocommerce_shipping_zones: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
gg_woocommerce_shipping_zone_locations: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
gg_woocommerce_shipping_zone_methods: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
gg_woocommerce_payment_tokens: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
gg_woocommerce_payment_tokenmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
gg_woocommerce_log: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
gg_actionscheduler_actions: Daten: 0.02MB + Index: 0.11MB + Engine InnoDB
gg_actionscheduler_claims: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
gg_actionscheduler_groups: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
gg_actionscheduler_logs: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
gg_commentmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
gg_comments: Daten: 0.02MB + Index: 0.08MB + Engine InnoDB
gg_links: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
gg_options: Daten: 2.48MB + Index: 0.06MB + Engine InnoDB
gg_postmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
gg_posts: Daten: 0.02MB + Index: 0.06MB + Engine InnoDB
gg_term_relationships: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
gg_term_taxonomy: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
gg_termmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
gg_terms: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
gg_usermeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
gg_users: Daten: 0.02MB + Index: 0.05MB + Engine InnoDB
gg_wc_admin_note_actions: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
gg_wc_admin_notes: Daten: 0.06MB + Index: 0.00MB + Engine InnoDB
gg_wc_category_lookup: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
gg_wc_customer_lookup: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
gg_wc_download_log: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
gg_wc_order_coupon_lookup: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
gg_wc_order_product_lookup: Daten: 0.02MB + Index: 0.06MB + Engine InnoDB
gg_wc_order_stats: Daten: 0.02MB + Index: 0.05MB + Engine InnoDB
gg_wc_order_tax_lookup: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
gg_wc_product_attributes_lookup: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
gg_wc_product_download_directories: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
gg_wc_product_meta_lookup: Daten: 0.02MB + Index: 0.09MB + Engine InnoDB
gg_wc_rate_limits: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
gg_wc_reserved_stock: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
gg_wc_tax_rate_classes: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
gg_wc_webhooks: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
page: 7
post: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ❌Fehlermeldungen sollten Besuchern nicht angezeigt werden.

### Active Plugins (1) ###

WooCommerce: von Automattic – 7.6.1

### Inactive Plugins (2) ###

WooCommerce Payments: von Automattic – 5.8.1
WooCommerce Shipping & Tax: von WooCommerce – 2.2.4

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: right_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
Order datastore: WC_Order_Data_Store_CPT

### WC Pages ###

Shop-Basis: #6 - /shop/
Warenkorb: #7 - /warenkorb/
Kasse: #8 - /kasse/
Mein Konto: #9 - /mein-konto/
Allgemeine Geschäftsbedingungen: ❌ Seite nicht festgelegt

### Theme ###

Name: Storefront
Version: 4.2.0
Author URL: https://woocommerce.com/
Child Theme: ❌ – Wenn du WooCommerce mit einem Theme
das du nicht selbst entwickelt hast
an eigene Bedürfnisse anpasst
dann empfehlen wir die Verwendung eines Child-Themes. Siehe Wie du ein Child-Theme erstellst (engl.)

WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
multichannel-marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: block-editor-feature-enabled
minified-js
new-product-management-experience
product-variation-management
settings

Daily Cron: ✔ Next scheduled: 2023-05-10 09:28:42 +02:00
Options: ✔
Notes: 62
Onboarding: completed

### Action Scheduler ###

Abgeschlossen: 5
Oldest: 2023-05-09 11:29:53 +0200
Newest: 2023-05-09 11:45:17 +0200

Fehlgeschlagen: 1
Oldest: 2023-05-09 11:47:17 +0200
Newest: 2023-05-09 11:47:17 +0200

Ausstehend: 1
Oldest: 2023-05-10 11:29:53 +0200
Newest: 2023-05-10 11:29:53 +0200

### Status report information ###

Generated at: 2023-05-09 13:29:22 +02:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by wdfee** [comment]
To debug this I checked file /[…]/plugins/woocommerce/src/Internal/Admin/Translations.php. These are lines 78 to 84:

```
$chunk_data    = \json_decode( $file_contents, true );

if ( empty( $chunk_data ) ) {
	continue;
}

$reference_file = $chunk_data['comment']['reference'];
```

The code seems to expect either an empty array and therefore pass the foreach loop to the next, or an array that must have a ‘comment’ key on first level and a ‘reference’ key on second level. 
To debug the $chunk_data array, I changed lines 80-82 to:

```
if(!empty($chunk_data['comment'])) {
    echo '<pre>'; print_r($chunk_data['comment']['reference']); echo '</pre>'; 
} else {
    echo '<pre>'; print_r($chunk_data); echo '</pre>'; 
}
```

In the output I see that the second if statement throws an array without any 'comment' or 'reference' key, so without a reference_file.
I put the output on gist, because it's too long for here. 
First all referenced files (first if statement): https://gist.github.com/wdfee/9e9e64bfca7ed31963f6fe7c3bced254
and then the Array in which the file reference is missing (second if statement): https://gist.github.com/wdfee/b5ab68a51fd7a44701e7a0515556bd76

**Comment by Babylon1999** [comment]
Same story in zd-6320829

I can't replicate this with [LocalWP](https://localwp.com/) so I advised them to try it out. :\

**Comment by ghulmarduk** [comment]
Any news on this topic? I tried setting up WooCommerce on a fresh & clean WP installation (German) and am getting the above described error right from the start. No chance for me to use WooCommerce under these circumstances.

**Comment by moon0326** [comment]
✋  Sorry for the delay in response.

I was able to reproduce it in my local setup. The warning only appears when you enable `WP_DEBUG_DISPLAY` config and when you deactivate and activate WooCommerce again (at least in my case).

It looks like the translation file is missing comment field. I've raised this issue in our internal i18n team to get more info. I'll keep everyone posted as I get more information.
